huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template_group.go

```diff
@@ -186,224 +187,325 @@ func ResourceTranscodingTemplateGroup() *schema.Resource {
+
+func buildVideosBodyParams(rawVideos []interface{}) []map[string]interface{} {
 	if len(rawVideos) == 0 {
 		return nil
 	}
 
-	videos := make([]mpc.VideoObj, len(rawVideos))
-	for i, rawVideo := range rawVideos {
-		video := rawVideo.(map[string]interface{})
-		videos[i] = mpc.VideoObj{
-			Width:   int32(video["width"].(int)),
-			Height:  int32(video["height"].(int)),
-			Bitrate: int32(video["bitrate"].(int)),
+	videos := make([]map[string]interface{}, 0, len(rawVideos))
+	for _, video := range rawVideos {
+		video := video.(map[string]interface{})
+		params := map[string]interface{}{
+			"width":   video["width"],
+			"height":  video["height"],
+			"bitrate": video["bitrate"],
 		}
+		videos = append(videos, params)
 	}
 
-	return &videos
+	return videos
 }
 
-func buildVideoCommonOpts(rawVideoCommon []interface{}) *mpc.VideoCommon {
-	if len(rawVideoCommon) != 1 {
+func buildAudioBodyParams(rawAudio []interface{}) map[string]interface{} {
+	if len(rawAudio) == 0 {
 		return nil
 	}
 
-	videoCommon := rawVideoCommon[0].(map[string]interface{})
-	videoCommonOpts := mpc.VideoCommon{
-		Codec:              utils.Int32(int32(videoCommon["codec"].(int))),
-		Profile:            utils.Int32(int32(videoCommon["profile"].(int))),
-		Level:              utils.Int32(int32(videoCommon["level"].(int))),
-		Preset:             utils.Int32(int32(videoCommon["quality"].(int))),
-		MaxIframesInterval: utils.Int32(int32(videoCommon["max_iframes_interval"].(int))),
-		BframesCount:       utils.Int32(int32(videoCommon["max_consecutive_bframes"].(int))),
-		FrameRate:          utils.Int32(int32(videoCommon["fps"].(int))),
-		BlackCut:           utils.Int32(int32(videoCommon["black_bar_removal"].(int))),
-		OutputPolicy:       buildVideoCommonOutputPolicyOpts(videoCommon["output_policy"].(string)),
-	}
-
-	return &videoCommonOpts
+	audio := rawAudio[0].(map[string]interface{})
+	audioParams := map[string]interface{}{
+		"output_policy": utils.ValueIgnoreEmpty(audio["output_policy"]),
+		"codec":         audio["codec"],
+		"sample_rate":   audio["sample_rate"],
+		"bitrate":       utils.ValueIgnoreEmpty(audio["bitrate"]),
+		"channels":      audio["channels"],
+	}
+
+	return audioParams
 }
 
-func buildVideoCommonOutputPolicyOpts(outputPolicy string) *mpc.VideoCommonOutputPolicy {
-	if outputPolicy == "" {
+func buildVideoCommonBodyParams(rawVideoCommon []interface{}) map[string]interface{} {
+	if len(rawVideoCommon) == 0 {
 		return nil
 	}
 
-	var outputPolicyOpts mpc.VideoCommonOutputPolicy
-	switch outputPolicy {
-	case "discard":
-		outputPolicyOpts = mpc.GetVideoCommonOutputPolicyEnum().DISCARD
-	case "transcode":
-		outputPolicyOpts = mpc.GetVideoCommonOutputPolicyEnum().TRANSCODE
-	default:
-		log.Printf("[WARN] output_policy invalid: %s", outputPolicy)
-		return nil
+	videoCommon := rawVideoCommon[0].(map[string]interface{})
+	videoCommonParams := map[string]interface{}{
+		"output_policy":        utils.ValueIgnoreEmpty(videoCommon["output_policy"]),
```

Comment:
Please carefully compare the original logic to see whether the default value is nil or empty.